crates/cargo-contract/src/cmd/build.rs

```diff
@@ -145,6 +153,15 @@ impl BuildCommand {
             false => OutputType::HumanReadable,
         };
 
+        let image = if self.verifiable {
+            match &self.image {
+                Some(i) => Some(ImageVariant::Custom(i.clone())),
+                None => Some(ImageVariant::Default),
+            }
+        } else {
+            None
+        };
```

Comment:
Yes here self.verifiable == false && self.image.is_some() is invalid. In any case there is some duplication of mapping from `Option<ImageVariant>` to `ImageVariant` which only should happen in one place.

Reply:
The duplication is of the match statement, conversion could/should happen all in one place `Option<String>` -> `ImageVariant`. At the moment we match `None` -> `None`, then the `None` gets translated to a `ImageVariant::Default`.